unit-tests/src/test/scala/scala/scalanative/libc/CComplexTest.scala

```diff
@@ -170,7 +170,7 @@ class CComplexTest {
   @Test def testCpowf(): Unit = {
     Zone { implicit z =>
       assertEqualsComplexF(cpowf(tf, tf, buff),
-                           res(toFloat(0x3e8c441e), toFloat(0x3f156d6a)))
+                           res(toFloat(0x3e8c4420), toFloat(0x3f156d6a)))
```

Comment:
I'll bight, why the change?  Seems to be a change by 2. A bit being set?

Reply:
A couple of comments on this change.
1. The results where generated on macOS but work on Linux as well.
2. Currently we are only doing CI on Linux but I think we need to consider supporting macOS and BSD if we can.
3. If looks like there is a small additional error on this test so maybe we can add something like the following:
```Scala
def isAlmostEqual(act: Float, exp: Float, ulpFactor: Int): Boolean = ...
// replace 2 in the function with ulpFactor
// Also add the same with a default to the following and make appropriate changes
def assertEqualsComplexF(act: Ptr[CFloatComplex],
                           exp: Ptr[CFloatComplex], ulpFactor = 2): Unit =...
// pass the ulpFactor through
```

Then for the test in question you can try `3` and leave a comment that BSD is off by three compared to macOS and Linux off by 2.

Reply:
From what you say I understand that the literal numbers were generated as copied as-is from macOS. There is an identical test with `Cpow` that uses `double` instead of `float`, that I assume produces more accurated results. What I did was to take the complex double and convert it to a complex float. This is how I generated the number, it's not the result from FreeBSD, so I think it's a more accurate number (supported by the fact that  the CI on Linux didn't break). If macOS is fine too I don't think we need to change `2` with `3`.

Reply:
That is fine if it works on macOS if you put a comment with the "// old value macOS, new value changed for BSD". The code generation is below so it can be repeated and studied - not sure if it is the best way or not.

The double code uses a `ulpFactor` of 1 or just `epsilon` vs 2 for float.

Reply:
I've run the code. These are the FreeBSD values:
```
cpowf   -> toFloat(0x3e8c4421), toFloat(0x3f156d69)
cpow    -> toDouble("3fd18884016cf327"), toDouble("3fe2adad36b098aa")
```
So as you can see the double values are the same, while the float are different from both the original ones and the patched ones. As I said, the float value has been generated by converting the double value, so it should be more accurate than any direct output of `cpowf`. I cannot test it on macOS.

Reply:
It should work on macOS, too, the `assertEqualsComplexF` function returns `true` when I pass the original numbers.

Reply:
ok, that sounds good.